corehq/motech/repeaters/tasks.py

```diff
@@ -170,6 +174,7 @@ def _process_repeat_record(repeat_record):
             # with the intent of avoiding clumping and spreading load
             repeat_record.postpone_by(random.uniform(*RATE_LIMITER_DELAY_RANGE))
         elif repeat_record.is_queued():
+            report_repeater_attempt(repeat_record.domain)
```

Comment:
To me, this is the crux of this change. Having something that reports _before_ we attempt to forward is very valuable when dealing with slow endpoints, since by the time the slow endpoints return and we report wait times, our rate limiting code is taking action too late. By reporting before, we can prevent too many workers from actually attempting to forward to a slow endpoint.